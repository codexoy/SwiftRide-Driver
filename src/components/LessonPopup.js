import React from "react"
import { StyleSheet, View, Text, TouchableOpacity } from "react-native"
import Modal from "react-native-modal"
import { strings } from "../i18n"
import Hours from "./Hours"
import moment from "moment"
import { fullButton, NAME_LENGTH } from "../consts"
import { getUserImage } from "../actions/utils"
import FastImage from "react-native-fast-image"

export default class LessonPopup extends React.Component {
	constructor(props) {
		super(props)
		this.navigateToProfile = this.navigateToProfile.bind(this)
		this.navigateToLesson = this.navigateToLesson.bind(this)
	}

	navigateToLesson = () => {
		this.props.onPress(this.props.item)
		this.props.navigation.navigate("Lesson", {
			lesson: this.props.item
		})
	}

	navigateToProfile = () => {
		this.props.onPress(this.props.item)
		this.props.navigation.navigate("StudentProfile", {
			student: this.props.item.student
		})
	}

	render() {
		const { item } = this.props
		if (!item) return null
		let student = strings("teacher.no_student_applied")
		let image
		if (item.student) {
			const name = item.student.user.name.slice(0, NAME_LENGTH)
			student = `${strings("teacher.home.lesson_number")} ${
				item.lesson_number
			} - ${name}`
			if (item.student.user.image) {
				image = (
					<FastImage
						style={styles.image}
						source={{
							uri: getUserImage(item.student.user)
						}}
					/>
				)
			}
		}
		let meetup = strings("not_set")
		if (item.meetup_place) meetup = item.meetup_place.name
		let dropoff = strings("not_set")
		if (item.dropoff_place) dropoff = item.dropoff_place.name
		let approved
		if (!item.is_approved) {
			approved = (
				<Text style={{ color: "red" }}>
					({strings("not_approved")})
				</Text>
			)
		}
		return (
			<Modal
				isVisible={this.props.visible}
				onBackdropPress={() => this.props.onPress(item)}
				animationIn="pulse"
				animationOut="fadeOut"
			>
				<View style={styles.popup} testID={this.props.testID}>
					<TouchableOpacity onPress={this.navigateToProfile}>
						{image}
					</TouchableOpacity>
					<Text style={styles.title}>{student}</Text>
					<Text style={styles.approved}>{approved}</Text>
					<View style={styles.row}>
						<View style={styles.column}>
							<Text style={styles.titles}>
								{strings("teacher.new_lesson.date")}
							</Text>
							<Text style={styles.texts}>
								{moment.utc(item.date).format("DD-MM-YYYY")}
							</Text>
						</View>
						<View style={{ ...styles.column, marginLeft: 2 }}>
							<Text style={styles.titles}>
								{strings("teacher.new_lesson.hour")}
							</Text>
							<Hours
								style={styles.texts}
								duration={item.duration}
								date={item.date}
							/>
						</View>
					</View>
					<View style={styles.row}>
						<View style={styles.column}>
							<Text style={styles.titles}>
								{strings("teacher.new_lesson.meetup")}
							</Text>
							<Text style={styles.texts}>{meetup}</Text>
						</View>
						<View style={{ ...styles.column, marginLeft: 2 }}>
							<Text style={styles.titles}>
								{strings("teacher.new_lesson.dropoff")}
							</Text>
							<Text style={styles.texts}>{dropoff}</Text>
						</View>
					</View>
					<TouchableOpacity
						underlayColor="#ffffff00"
						onPress={this.navigateToLesson}
						style={styles.button}
					>
						<View testID="editLessonButton">
							<Text style={styles.buttonText}>
								{strings("edit_lesson")}
							</Text>
						</View>
					</TouchableOpacity>
				</View>
			</Modal>
		)
	}
}

const styles = StyleSheet.create({
	popup: {
		flex: 1,
		maxHeight: 320,
		backgroundColor: "#fff",
		padding: 26,
		alignSelf: "center",
		width: 280,
		alignContent: "center",
		borderRadius: 4
	},
	image: {
		marginTop: -80,
		width: 76,
		height: 76,
		borderRadius: 38,
		alignSelf: "center"
	},
	title: {
		fontWeight: "bold",
		fontSize: 18,
		alignSelf: "center",
		marginTop: 16
	},
	approved: { alignSelf: "center", marginTop: 6 },
	row: {
		flex: 1,
		flexDirection: "row",
		alignItems: "flex-start",
		marginTop: 20,
		maxHeight: 60
	},
	column: {
		flexDirection: "column",
		justifyContent: "flex-start",
		width: 140
	},
	titles: {
		fontWeight: "bold",
		fontSize: 14,
		color: "gray",
		alignSelf: "flex-start"
	},
	texts: { fontSize: 18, marginTop: 6, alignSelf: "flex-start" },
	button: { ...fullButton, width: 280 },
	buttonText: {
		fontWeight: "bold",
		color: "#fff",
		fontSize: 20
	}
})
