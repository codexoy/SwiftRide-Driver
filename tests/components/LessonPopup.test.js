import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import LessonPopup from "../../src/components/LessonPopup"

describe("LessonPopup", () => {
	test("view renders correctly", () => {
		const item = {
			date: "Sun, 23 Jun 2019 10:09:25 GMT",
			duration: "40",
			student: { user: { name: "test" } }
		}
		const tree = renderer
			.create(
				<LessonPopup
					visible={true}
					item={item}
					onPress={jest.fn()}
					testID="test"
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
