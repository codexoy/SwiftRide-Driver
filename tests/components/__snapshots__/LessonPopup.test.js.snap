// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonPopup view renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "flex": 1,
          "maxHeight": 320,
          "padding": 26,
          "width": 280,
        }
      }
      testID="test"
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginTop": 16,
          }
        }
      >
        שיעור מספר undefined - test
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "marginTop": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "red",
            }
          }
        >
          (
          לא מאושר
          )
        </Text>
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "marginTop": 20,
            "maxHeight": 60,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": 140,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "gray",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            תאריך
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 18,
                "marginTop": 6,
              }
            }
          >
            23-06-2019
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 2,
              "width": 140,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "gray",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            שעה
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 18,
                "marginTop": 6,
              }
            }
          >
            10:09
             - 
            10:49
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "marginTop": 20,
            "maxHeight": 60,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": 140,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "gray",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            איסוף
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 18,
                "marginTop": 6,
              }
            }
          >
            לא נקבע
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 2,
              "width": 140,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "gray",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            הורדה
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 18,
                "marginTop": 6,
              }
            }
          >
            לא נקבע
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(12,116,244)",
            "borderRadius": 0,
            "bottom": 0,
            "elevation": 8,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "rgb(12,116,244)",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 16,
            "width": 280,
          }
        }
      >
        <View
          testID="editLessonButton"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            עריכת שיעור
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
